File: pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import java.util.List;

import com.pinyougou.pojogroup.Cart;

/**
 * 购物车服务接口
 * @author Administrator
 *
 */
public interface CartService {

	/**
	 * 添加商品到购物车
	 * @param cartList 购物车列表
	 * @param itemId   SKU的ID
	 * @param num	   商品数量
	 * @return	完成操作后的购物车列表
	 */
	public List<Cart>  addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);

	List<Cart> findCartListFromRedis(String userName);

	void saveCartListToRedis(String username, List<Cart> cartList);
}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
//购物车服务层
app.service('cartService',function($http){
	
	//查询购物车列表
	this.findCartList=function(){
		return $http.get('cart/findCartList.do');
	}


	//添加商品到购物车
	this.addGoodsToCartList=function(itemId,num){
		return $http.get('cart/addGoodsToCartList.do?itemId='+itemId+'&num='+num);
	}


	//在订单详情页-点击提交订单
	this.submitOrder = function(order){
		return $http.post('order/add.do',order);
    }

});